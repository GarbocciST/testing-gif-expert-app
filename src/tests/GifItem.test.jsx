import { render, screen } from "@testing-library/react"
import { GifItem } from "../components/GifItem"

describe('Pruebas en GiftItem', () => { 

  const title = 'one punch';
  const url = 'https://media0.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=26ecf56anf7vnosp5b0g7gnwaeo8vynqkwpues9v1515o4k3&rid=giphy.gif&ct=g%22';

  test('Debe hacer match con el snapshot', () => { 

    const {container} = render(<GifItem title={title} url={url} />)

    expect(container).toMatchSnapshot();

  })

  test('Debe haber un parrafo con titulo', () => { 

    render(<GifItem title={title} url={url} />)

    expect(screen.getByText(title)).toBeTruthy();

  })
  
  test('Debe haber url y title en img', () => { 

    render(<GifItem title={title} url={url} />)

    // screen.debug();

    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(title);
    const {src, alt} = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);

  })
  

})