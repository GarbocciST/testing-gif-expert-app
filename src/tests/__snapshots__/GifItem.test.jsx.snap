// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="one punch"
      src="https://media0.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=26ecf56anf7vnosp5b0g7gnwaeo8vynqkwpues9v1515o4k3&rid=giphy.gif&ct=g%22"
    />
    <p>
      one punch
    </p>
  </div>
</div>
`;
